### ISSUE TITLE
No pipe/ pulseaudio backend on   v0.2.24 spotifyd-linux-full.tar.gz

### ISSUE THREAD

**Issue by oschade** [Spotifyd/spotifyd#589]
**Description**
I downloaded the spotifyd-linux-full.tar.gz of version 0.2.24 and run "./spotifyd -h":

`....
-b, --backend <string>                         The audio backend to use [possible values: alsa]
....
`

There is no pulseaudio backend, just alsa.

**Comment by awerlang** [comment]
It's not a bug in the program but in the continuous deployment. There has been a previous attempt to raise this issue but then a pull request closed it on error, as it didn't fixed the issue.

```yaml
- artifact_type: 'full'
  feature: 'dbus_keyring,dbus_mpris'  # Full version has all OS compatible features enabled
```

This doesn't make sense. To avoid creating builds for each combination of features, it would be better to define builds by capabilities:

- headless: can only be used as a Spotify Connect receiver (current slim) [1];
- dbus: doesn't need a separate Spotify device, can also be controlled by MPRIS (the current full);
- default: all features available for a given OS/arch (a new full, replaces macos-rodio);

[1] maybe it could be controlled with a system bus through ssh, but AFAIK it's not currently available, and in that cause the debus build could be used.

**Comment by awerlang** [comment]
The pipe backend depends on #354, and it would be nice to have enabled in all builds.

**Comment by SirWindfield** [comment]
@awerlang I am open to merging a PR that changes the current artifacts! I am already pretty limited in time that I can put into OSS and I do need help to keep this software up-to-date by the community :)

IMO, the following "rules" should be followed though:
- provide a slim version with only the necessary features enabled to make spotifyd work (no extras like dbus_keyring).
- provide a default artifact that has extras enabled + the most common audio backend for that platform.
- provide an `all` articact that has everything enabled that makes sense on a given platform. for example, this would still imply no `alsa_backend` on macOS for example.
